.github/workflows/lint.yml

```diff
@@ -0,0 +1,44 @@
+name: "Lint Checks"
+
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  # Basic smoke test that doesn't require secrets
+  lint-check:
+    name: Smoke Test (${{ matrix.os }})
+    
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```

Comment:
We may want to go to 22 right off the bat here since 20 goes out of security support in ~7 months. We'll need to also update action.yml / the [types dependency](https://github.com/tailscale/github-action/pull/199/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R23) if we do. We could jump even further to 24, but think there are still runners (internally or otherwise) that won't have support for 24 just yet.

Might also be good to note for ourselves that bumping this typically requires a major version bump in the action itself as well (e.g., see v5 of actions/checkout [here](https://github.com/actions/checkout/releases/tag/v5.0.0)).